src/main/java/edu/harvard/iq/dataverse/engine/command/impl/MergeInAccountCommand.java

```diff
@@ -0,0 +1,162 @@
+import edu.harvard.iq.dataverse.RoleAssignment;
+import edu.harvard.iq.dataverse.UserNotification;
+import edu.harvard.iq.dataverse.authorization.AuthenticatedUserLookup;
+import edu.harvard.iq.dataverse.authorization.providers.builtin.BuiltinUser;
+import edu.harvard.iq.dataverse.authorization.users.ApiToken;
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.confirmemail.ConfirmEmailData;
+import edu.harvard.iq.dataverse.engine.command.AbstractVoidCommand;
+import edu.harvard.iq.dataverse.engine.command.CommandContext;
+import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
+import edu.harvard.iq.dataverse.engine.command.RequiredPermissions;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+import edu.harvard.iq.dataverse.engine.command.exception.IllegalCommandException;
+import edu.harvard.iq.dataverse.search.IndexResponse;
+import edu.harvard.iq.dataverse.search.savedsearch.SavedSearch;
+import edu.harvard.iq.dataverse.workflows.WorkflowComment;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Merges one account into another.
+ * 
+ * @author matthew
+ */
+
+@RequiredPermissions({})
+public class MergeInAccountCommand extends AbstractVoidCommand {
+    final AuthenticatedUser consumedAU;
+    final BuiltinUser consumedBU;
+    final AuthenticatedUser ongoingAU;
+
+    private static final Logger logger = Logger.getLogger(MergeInAccountCommand.class.getCanonicalName());
+    
+    public MergeInAccountCommand(DataverseRequest createDataverseRequest, AuthenticatedUser consumedAuthenticatedUser, 
+            BuiltinUser consumedBuiltinUser, AuthenticatedUser ongoingAU) {
+        super(
+            createDataverseRequest,
+            (DvObject) null
+        );
+        consumedAU = consumedAuthenticatedUser;
+        consumedBU = consumedBuiltinUser;
+        this.ongoingAU = ongoingAU;
+    }
+    
+    @Override
+    protected void executeImpl(CommandContext ctxt) throws CommandException {
+        
+        List<RoleAssignment> baseRAList = ctxt.roleAssignees().getAssignmentsFor(ongoingAU.getIdentifier());
+        List<RoleAssignment> consumedRAList = ctxt.roleAssignees().getAssignmentsFor(consumedAU.getIdentifier());
+        
+        for(RoleAssignment cra : consumedRAList) {
+            if(cra.getAssigneeIdentifier().charAt(0) == '@') {
+                //This still needs to check if the same Role Assignment exists in the baseRAList and if so... what?
+                boolean willDelete = false;
+                for(RoleAssignment bra : baseRAList) {
+                    //Matching on the id not the whole DVObject as I'm suspicious of dvobject equality
+                    if( bra.getDefinitionPoint().getId().equals(cra.getDefinitionPoint().getId())
+                        && bra.getRole().equals(cra.getRole())) { //TODO: Match on privateurltoken as well?? What does this do?
```

Comment:
I'm happy to talk through the privateurltoken stuff if we'd like to remove this TODO from the code.

See also http://phoenix.dataverse.org/site/latest/site/apidocs/edu/harvard/iq/dataverse/privateurl/package-summary.html which starts with "The Private URL feature has been implemented as a specialized role assignment with an associated token that permits read-only access to the metadata and all files (regardless of if the files are restricted or not) of a draft version of a dataset."

And "A PrivateUrlUser is virtual in the sense that it does not have a row in the authenticateduser table. It exists so when a Private URL is enabled for a dataset, we can assign a read-only role ("member") to the identifier for the PrivateUrlUser." http://phoenix.dataverse.org/site/latest/site/apidocs/edu/harvard/iq/dataverse/authorization/users/PrivateUrlUser.html

In short, I don't think we need to worry about Private URLs for this pull request since they are not associated with the authenticateduser table. They're part of the dataset and shouldn't be touched with user accounts are merged.